### PR TITLE
Updating docs for torch.distributions

### PR DESCRIPTION
**Pull request by vishwakftw**
This closes https://github.com/probtorch/pytorch/issues/67. So far, the following has been done:

- [x] Add a mini paragraph
- [x] Add a sample snippet

### CODE REVIEW

File: torch/distributions/__init__.py [hunk 1]
```diff
@@ -28,6 +28,21 @@
...
+Another way to implement these stochastic/policy gradients would be to use the
+reparameterization trick from :meth:`~torch.distributions.Distribution.rsample`
+method, where the parameterized random variable can be defined as a parameterized
+deterministic function of a parameter-free random variable. The reparameterized sample
+is required to be differentiable. The code for implementing the pathwise estimation would
+be as follows::
+
+    probs = policy_network(state)
+    # NOTE: this is equivalent to what used to be called multinomial
+    m = Categorical(probs)
```
Comment (fritzo) [inline torch/distributions/__init__.py]: `Categorical` is not reparameterized; we should use an example with `.has_rsample` here, e.g.

```py
params = policy_network(state)
action = Normal(**params).rsample()
next_state, reward = env.step(action)  # assume enf.step is differentiable
loss = -reward
loss.backward()
```

File: torch/distributions/__init__.py [hunk 2]
```diff
@@ -28,6 +28,21 @@
...
     loss.backward()
+
+Another way to implement these stochastic/policy gradients would be to use the
+reparameterization trick from :meth:`~torch.distributions.Distribution.rsample`
+method, where the parameterized random variable can be defined as a parameterized
+deterministic function of a parameter-free random variable. The reparameterized sample
+is required to be differentiable. The code for implementing the pathwise estimation would
+be as follows::
+
+    probs = policy_network(state)
```
Comment (fritzo) [inline torch/distributions/__init__.py]: these aren't probs, these are params. I would either write
```py
loc, scale = policy_network(state)
m = Normal(loc, scale)
```
or sweep them into a `params`
```py
params = policy_network(state)
m = Norma(*params)
```

File: torch/distributions/__init__.py [hunk 3]
```diff
@@ -28,6 +28,21 @@
...
+    m = Normal(**probs)
+    # any distribution with has_rsample == True could work based on the application
```
Comment (fritzo) [inline torch/distributions/__init__.py]: nit: I'd spell this `.has_rsample == True` where the dot emphasizes that `.has_rsample` is an attribute.

**Comment by vishwakftw** [comment]
I have fixed the nits suggested, sorry about those. Will close and send upstream.

**Pull request closed without merging**